radify/radiian
# NOMERGE [#99694710] Make provision.sh executable

## NOMERGE [#99694710] Make provision.sh executable
## Failed to add optional parameter to write()
- In order to make 'provision.sh' executable, I choose to use
  the optional paramenter in 'fs.write'.
- In this case, 'fs.write' is wrapped in a Promise, 'write'
  which is in turn wrapped in a function, 'getWriter'
- I added an optional parameter to 'getWriter' and passed it
  to 'write'.
- The 'parseInt' part is necessary because octal numbers are
  not allowed in strict mode, so they must be passed as strings
  to parseInt.
- 'getWriter('ansible/provision.sh', '0755')' still results in
  permissions '0644' for that file. I don't know why.
- One option to get out of this mess would be not to modify 'write'
  and 'getWriter', but rather create a new function like this:
  `var chmod = Q.denodeify(fs.chmod)`. Then this could be used
  where necessary.
## Test script
- Run `node src/cli.js`
  - Use `AKIAIOSFODNN7EXAMPLE` for the AWS key.
  - Use `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` for the secret key.
- Run `ls -al ansible/provision.sh`

**Assert that** `provision.sh`'s permissions are 0755.
- Note: If you'd like to see the permissions in octal, run `perl -e 'printf "%04o %s\n", (stat)[2] & 07777, $_ for @ARGV' ansible/provision.sh`
